src/decode.rs

```diff
@@ -377,27 +377,63 @@ mod tests {
     use crate::bitwriter::BitWriter;
     use crate::encode;
     use crate::exec::BitMachine;
+    use crate::jet::Core;
+    use bitcoin_hashes::hex::ToHex;
+
+    fn assert_program_succeeds<J: Jet>(prog_bytes: &[u8]) -> Rc<CommitNode<J>> {
```

Comment:
`assert_program_serialization_succeeds` or `assert_program_serializable` might be better identifiers.